<launch>

    <arg name="namespace" default="sam"/>

    <!--
    <node name="pid_setpoint_static" pkg="sam_loop_test" type="pid_setpoint_static" ns="LCG_trim">
      <param name="setpoint_topic" value="/pitch_setpoint" />
      <param name="setpoint" value="0.15" />
    </node>

    <node name="pid_setpoint_static" pkg="sam_loop_test" type="pid_setpoint_static" ns="VBS_depth">
      <param name="setpoint_topic" value="/depth_setpoint" />
      <param name="setpoint" value="0.5" />
    </node>
    -->

    <node name="pid_tf_listener" pkg="sam_loop_test" type="pid_tf_listener" output="screen" >
	    <param name="odom_frame" value="$(arg namespace)_odom"/>
	    <param name="base_frame" value="$(arg namespace)/base_link"/>
	    <param name="world_frame" value="$(arg namespace)_odom"/>
    </node>

    <node name="pid_actuator_tcg" pkg="sam_loop_test" type="pid_actuator_tcg" ns="TCG_roll" >
      <param name="topic_from_controller_1" value="tcg_control_action" />
      <param name="topic_from_controller_2" value="tcg_control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="0.01" />
      <remap from="/actuator_input" to="/uavcan_tcg_command"/>
    </node>


    <node name="controller" pkg="pid" type="controller" ns="TCG_roll" output="screen" >
      <param name="Kp" value="5.0" />
      <param name="Ki" value="5.0" />
      <param name="Kd" value="2.0" />
      <param name="upper_limit" value="3.1415" />
      <param name="lower_limit" value="-3.1415" />
      <param name="windup_limit" value="10" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <param name="topic_from_plant" value="/roll_feedback" />
      <param name="topic_from_controller" value="tcg_control_action" />
      <remap from="setpoint" to="/roll_setpoint" />
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     </node>


</launch>
